go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -76,6 +76,90 @@ func newEgressFirewallRule(rawEgressFirewallRule egressfirewallapi.EgressFirewal
 	return efr, nil
 }
 
+// This function is used to sync egress firewall setup. It does two "cleanups"
+
+// - 	Cleanup the old implementation (using LRP) in local GW mode -> new implementation (using ACLs) local GW mode
+//  	For this it just deletes all LRP setup done for egress firewall
+
+// -	Cleanup the new local GW mode implementation (using ACLs on the node switch) -> shared GW mode implementation (using ACLs on the join switch)
+//  	For this it just deletes all ACL setup done for egress firewall on the node switches
+
+// NOTE: Utilize the fact that we know that all egress firewall related setup must have a priority: types.MinimumReservedEgressFirewallPriority <= priority <= types.EgressFirewallStartPriority
+func (oc *Controller) syncEgressFirewall(egressFirwalls []interface{}) {
```

Comment:
perhaps a follow up you may need to check if its possible to upgrade from "old implementation local gw" -> "shared gw" and see if this would handle that scenario.

Reply:
thanks